tests/v2/validation/terratest/cluster/kubernetes_upgrade_test.go

```diff
@@ -0,0 +1,36 @@
+package terratest
+
+import (
+	"testing"
+
+	cleanup "github.com/rancher/rancher/tests/terratest/functions/cleanup"
+	terratest "github.com/rancher/rancher/tests/terratest/functions/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type KubernetesUpgradeTestSuite struct {
+	suite.Suite
+}
+
+func (r *KubernetesUpgradeTestSuite) TestKubernetesUpgrade() {
+	r.T().Parallel()
+
+	// set up func
+	terraformOptions := terratest.Setup(r.T())
+
+	// clean up
+	defer cleanup.Cleanup(r.T(), terraformOptions)
```

Comment:
[clarification]:
does terratest teardown based on the config's `rancher: cleanup` value, or is there a way to prevent the resources from terratest from being destroyed / cleaned up?

Reply:
Yes, exactly.  Depending on the boolean value of `cleanup` from Rancher config, the resources will be destroyed or not, following the test completion.